src/url.js

```diff
@@ -650,3 +650,18 @@ export function checkCorsUrl(url) {
 export function tryDecodeUriComponent(component, opt_fallback) {
   return tryDecodeUriComponent_(component, opt_fallback);
 }
+
+/**
+ * Adds the path to the given url.
+ * @param {string} url
+ * @param {string} path
+ * @return {string}
+ */
+export function appendPathToUrl(url, path) {
+  if (!a) {
+    a = /** @type {!HTMLAnchorElement} */ (self.document.createElement('a'));
```

Comment:
@choumx knows better, but I think you might have to accept the anchor as a parameter as you can't use `self`.

Reply:
Yea, note that the only other usage is in a deprecated function. 

We also have a URL service for parsing URLs. See `url-impl.js` and `Services.urlForDoc`. You can use that for parsing a string into a URL object, and change this function to accept a URL object instead.

Reply:
Those approaches required me to whitelist the reactions component to use url-impl or parseUrlWithA, but they were all essentially just getting an a tag and setting the href, so I changed the function to pass in the a tag directly to appendPathToUrl, lmk if this would work

Reply:
`Services.urlForDoc` in `services.js` is the right way to get a service object.

`parseUrlWithA` is restricted precisely to avoid callers from creating their own anchor elements. :)

Reply:
Thanks for the more detailed explanation, fixed it on upcoming commit